client/files.go

```diff
@@ -587,16 +587,21 @@ func (client *Client) Pull(opts *PullOptions) error {
 	}
 
 	mr := multipart.NewReader(resp.Body, params["boundary"])
-	filesPart, err := mr.NextPart()
+	firstPart, err := mr.NextPart()
 	if err != nil {
 		return fmt.Errorf("cannot decode multipart payload: %w", err)
 	}
-	defer filesPart.Close()
+	defer firstPart.Close()
 
-	if filesPart.FormName() != "files" {
-		return fmt.Errorf(`expected first field name to be "files", got %q`, filesPart.FormName())
+	// Check if we got "response" first (error case) or "files" (success case)
+	if firstPart.FormName() == "response" {
+		// Decode and process the response to extract the actual error
+		return processPullResponsePart(firstPart)
 	}
-	if _, err := io.Copy(opts.Target, filesPart); err != nil {
+	if firstPart.FormName() != "files" {
+		return fmt.Errorf(`expected first field name to be "files" or "response", got %q`, firstPart.FormName())
```

Comment:
I'd prefer we keep this error message the same as before. Because we shouldn't really get "response" except in case of error, which is handled above.
```suggestion
		return fmt.Errorf(`expected first field name to be "files", got %q`, firstPart.FormName())
```